src/request.spec.ts

```diff
@@ -313,6 +340,25 @@ describe('Request', () => {
       });
   });
 
+  it('fails when wrong HTTP verb is used', (done) => {
```

Comment:
Shouldn't this already fail validation rather than come through to the request execution logic?